torch/_inductor/codegen/cuda/cuda_template.py

```diff
@@ -26,13 +28,13 @@ class CUDATemplate(KernelTemplate):
     def __init__(
         self,
         name: str,
-        input_nodes: List[IRNode],
+        input_nodes: List[Layout | Buffer],
         layout: Layout,
         input_reorder: Optional[List[int]] = None,
     ):
         super().__init__(name)
         self.input_nodes = input_nodes
-        self.output_node = Buffer("buf_out", layout)
+        self.output_node: Layout | Buffer = Buffer("buf_out", layout)
```

Comment:
Yeah I think you're right, it's always a Buffer. I changed `TensorMeta.from_irnodes` to accept an immutable `Sequence[Buffer | Layout]` instead of a List to make this typecheck.